theano/gpuarray/tests/test_rng_mrg.py

```diff
@@ -165,4 +165,25 @@ def test_validate_input_types_gpuarray_backend():
 
 
 def test_f16_nonzero():
-    cpu_f16_nonzero(mode=mode, op_to_check=GPUA_mrg_uniform)
+    try:
+        # To have theano.shared(x
+        theano.compile.shared_constructor(gpuarray_shared_constructor)
+        cpu_f16_nonzero(mode=mode, op_to_check=GPUA_mrg_uniform)
+    finally:
+        theano.compile.shared_constructor(gpuarray_shared_constructor,
+                                          remove=True)
+
+
+def test_cpu_target_with_shared_variable():
+    srng = MRG_RandomStreams()
+    s = np.random.rand(2, 3).astype('float32')
+    x = gpuarray_shared_constructor(s, name='x')
+    y = srng.uniform(x.shape, target='cpu')
+    y.name = 'y'
+    z = (x * y).sum()
+    z.name = 'z'
+
+    fz = theano.function([], z, mode=mode)
+
+    nodes = fz.maker.fgraph.toposort()
+    assert not any([isinstance(node.op, GPUA_mrg_uniform) for node in nodes])
```

Comment:
This is a poor test since it doesn't run with `device=cuda` anyway.

Reply:
I force the shared variable x to be on the GPU and I passe the mode to compile with the GPU opt. So it cover our need. All GPUs tests are done like that.

But I see, for this particular tests, mimiking more device=cuda by having a chance that the state is on the GPU would make a better test. I'll update it like that. But this test was about the optimization, so it was covering the need.

I just pushed an update to the test for this.